ERROR - 2024-02-19 11:40:59 --> mysqli_sql_exception: Column 'idCliente' cannot be null in C:\xampp\htdocs\tu_taller\system\Database\MySQLi\Connection.php:292
Stack trace:
#0 C:\xampp\htdocs\tu_taller\system\Database\MySQLi\Connection.php(292): mysqli->query('INSERT INTO `ca...', 0)
#1 C:\xampp\htdocs\tu_taller\system\Database\BaseConnection.php(693): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `ca...')
#2 C:\xampp\htdocs\tu_taller\system\Database\BaseConnection.php(620): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `ca...')
#3 C:\xampp\htdocs\tu_taller\system\Database\BaseBuilder.php(1916): CodeIgniter\Database\BaseConnection->query('INSERT INTO `ca...', Array, false)
#4 C:\xampp\htdocs\tu_taller\app\Models\CalificarModel.php(44): CodeIgniter\Database\BaseBuilder->insert(Array)
#5 C:\xampp\htdocs\tu_taller\app\Controllers\Calificar.php(30): App\Models\CAlificarModel->InsertCalificacion('5', NULL, '1')
#6 C:\xampp\htdocs\tu_taller\system\CodeIgniter.php(904): App\Controllers\Calificar->registrarCalificacion()
#7 C:\xampp\htdocs\tu_taller\system\CodeIgniter.php(457): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Calificar))
#8 C:\xampp\htdocs\tu_taller\system\CodeIgniter.php(340): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#9 C:\xampp\htdocs\tu_taller\public\index.php(67): CodeIgniter\CodeIgniter->run()
#10 {main}
CRITICAL - 2024-02-19 11:40:59 --> Column 'idCliente' cannot be null
in SYSTEMPATH\Database\MySQLi\Connection.php on line 292.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(292): mysqli->query('INSERT INTO `calificacion` (`idTaller`, `idCliente`, `calificacion`) VALUES (\'5\', NULL, \'1\')', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(693): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `calificacion` (`idTaller`, `idCliente`, `calificacion`) VALUES (\'5\', NULL, \'1\')')
 3 SYSTEMPATH\Database\BaseConnection.php(620): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `calificacion` (`idTaller`, `idCliente`, `calificacion`) VALUES (\'5\', NULL, \'1\')')
 4 SYSTEMPATH\Database\BaseBuilder.php(1916): CodeIgniter\Database\BaseConnection->query('INSERT INTO `calificacion` (`idTaller`, `idCliente`, `calificacion`) VALUES (:idTaller:, :idCliente:, :calificacion:)', [...], false)
 5 APPPATH\Models\CalificarModel.php(44): CodeIgniter\Database\BaseBuilder->insert([...])
 6 APPPATH\Controllers\Calificar.php(30): App\Models\CAlificarModel->InsertCalificacion('5', null, '1')
 7 SYSTEMPATH\CodeIgniter.php(904): App\Controllers\Calificar->registrarCalificacion()
 8 SYSTEMPATH\CodeIgniter.php(457): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Calificar))
 9 SYSTEMPATH\CodeIgniter.php(340): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
10 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
ERROR - 2024-02-19 12:24:32 --> mysqli_sql_exception: Cannot add or update a child row: a foreign key constraint fails (`db_taller`.`cita`, CONSTRAINT `fk_cita_servicio1` FOREIGN KEY (`idServicio`) REFERENCES `servicio` (`idServicio`)) in C:\xampp\htdocs\tu_taller\system\Database\MySQLi\Connection.php:292
Stack trace:
#0 C:\xampp\htdocs\tu_taller\system\Database\MySQLi\Connection.php(292): mysqli->query('INSERT INTO `ci...', 0)
#1 C:\xampp\htdocs\tu_taller\system\Database\BaseConnection.php(693): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `ci...')
#2 C:\xampp\htdocs\tu_taller\system\Database\BaseConnection.php(620): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `ci...')
#3 C:\xampp\htdocs\tu_taller\system\Database\BaseBuilder.php(1916): CodeIgniter\Database\BaseConnection->query('INSERT INTO `ci...', Array, false)
#4 C:\xampp\htdocs\tu_taller\app\Models\CitaModel.php(33): CodeIgniter\Database\BaseBuilder->insert(Array)
#5 C:\xampp\htdocs\tu_taller\app\Controllers\Cita.php(33): App\Models\CitaModel->InsertCita('', '2024-02-22T12:2...', 0, '5', '9', 0)
#6 C:\xampp\htdocs\tu_taller\system\CodeIgniter.php(904): App\Controllers\Cita->solicitarCita()
#7 C:\xampp\htdocs\tu_taller\system\CodeIgniter.php(457): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Cita))
#8 C:\xampp\htdocs\tu_taller\system\CodeIgniter.php(340): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#9 C:\xampp\htdocs\tu_taller\public\index.php(67): CodeIgniter\CodeIgniter->run()
#10 {main}
CRITICAL - 2024-02-19 12:24:32 --> Cannot add or update a child row: a foreign key constraint fails (`db_taller`.`cita`, CONSTRAINT `fk_cita_servicio1` FOREIGN KEY (`idServicio`) REFERENCES `servicio` (`idServicio`))
in SYSTEMPATH\Database\MySQLi\Connection.php on line 292.
 1 SYSTEMPATH\Database\MySQLi\Connection.php(292): mysqli->query('INSERT INTO `cita` (`descripcion`, `fechaCita`, `estado`, `idTaller`, `idCliente`, `idServicio`) VALUES (\'\', \'2024-02-22T12:24\', 0, \'5\', \'9\', 0)', 0)
 2 SYSTEMPATH\Database\BaseConnection.php(693): CodeIgniter\Database\MySQLi\Connection->execute('INSERT INTO `cita` (`descripcion`, `fechaCita`, `estado`, `idTaller`, `idCliente`, `idServicio`) VALUES (\'\', \'2024-02-22T12:24\', 0, \'5\', \'9\', 0)')
 3 SYSTEMPATH\Database\BaseConnection.php(620): CodeIgniter\Database\BaseConnection->simpleQuery('INSERT INTO `cita` (`descripcion`, `fechaCita`, `estado`, `idTaller`, `idCliente`, `idServicio`) VALUES (\'\', \'2024-02-22T12:24\', 0, \'5\', \'9\', 0)')
 4 SYSTEMPATH\Database\BaseBuilder.php(1916): CodeIgniter\Database\BaseConnection->query('INSERT INTO `cita` (`descripcion`, `fechaCita`, `estado`, `idTaller`, `idCliente`, `idServicio`) VALUES (:descripcion:, :fechaCita:, :estado:, :idTaller:, :idCliente:, :idServicio:)', [...], false)
 5 APPPATH\Models\CitaModel.php(33): CodeIgniter\Database\BaseBuilder->insert([...])
 6 APPPATH\Controllers\Cita.php(33): App\Models\CitaModel->InsertCita('ROOTPATH\\public', '2024-02-22T12:24', 0, '5', '9', 0)
 7 SYSTEMPATH\CodeIgniter.php(904): App\Controllers\Cita->solicitarCita()
 8 SYSTEMPATH\CodeIgniter.php(457): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Cita))
 9 SYSTEMPATH\CodeIgniter.php(340): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
10 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
